File: app-java/src/main/java/tech/thdev/java_udemy_sample/view/main/MainActivity.java
package tech.thdev.java_udemy_sample.view.main;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import tech.thdev.java_udemy_sample.R;
import tech.thdev.java_udemy_sample.databinding.ActivityMainBinding;
import tech.thdev.java_udemy_sample.util.ActivityUtil;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding activityMainBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        activityMainBinding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(activityMainBinding.getRoot());

        activityMainBinding.toolbar.setTitle(R.string.app_name);
        setSupportActionBar(activityMainBinding.toolbar);

        ActivityUtil.replaceFragmentToActivity(getSupportFragmentManager(), MainFragment.getInstance(), R.id.frame_layout);
    }
}

File: app-kotlin/src/main/java/tech/thdev/kotlin_udemy_sample/view/main/MainActivity.kt
package tech.thdev.kotlin_udemy_sample.view.main

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import tech.thdev.kotlin_udemy_sample.R
import tech.thdev.kotlin_udemy_sample.databinding.ActivityMainBinding
import tech.thdev.kotlin_udemy_sample.util.replaceFragmentToActivity

class MainActivity : AppCompatActivity() {

    private lateinit var activityMainBinding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityMainBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(activityMainBinding.root)

        setSupportActionBar(activityMainBinding.toolbar)

        replaceFragmentToActivity(MainFragment.getInstance(), R.id.frame_layout)
    }
}
